can/zynqmp-can/zynqmp-can-test.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * Phoenix-RTOS
+ *
+ * ZynqMP CAN driver test application
+ *
+ * Copyright 2025 Phoenix Systems
+ * Author: Dariusz Sabala
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <termios.h>
+#include <stddef.h>
+#include <sys/types.h>
+#include <phoenix/msg.h>
+#include <sys/threads.h>
+#include <sys/msg.h>
+#include <signal.h>
+#include <unistd.h>
+
+#include <zynqmp-can-if.h>
+
+/* Macro to calculate the length of an array */
+#define ARRAY_LENGTH(array) (sizeof((array)) / sizeof((array)[0]))
+
+/* Structure to store statistics for received frames */
+typedef struct {
+	uint32_t receivedFramesCount; /**< Number of received frames */
+	zynqmp_canFrame lastFrame;    /**< Last received frame */
+} frameStats;
+
+/* Mutex for accessing frame statistics */
+static handle_t mutex_stats_access;
+
+/* Array to store statistics for each CAN ID */
+static frameStats frame_stats[ZYNQMP_CAN_ID_MAX] = { 0 };
+
+/* Flag to control the main loop */
+static int run = 1;
+
+/**
+ * Waits for a command input for a specified number of seconds.
+ */
+static int waitcmd(unsigned int secs)
+{
+	struct timeval left;
+	fd_set fds;
+
+	left.tv_sec = (time_t)secs;
+	left.tv_usec = 0;
+	fflush(stdout);
+	FD_ZERO(&fds);
+	FD_SET(STDIN_FILENO, &fds);
+	select(STDIN_FILENO + 1, &fds, NULL, NULL, &left);
+
+	if (FD_ISSET(STDIN_FILENO, &fds))
+		return getchar();
+
+	return 0;
+}
+
+/**
+ * Decodes and prints the contents of a CAN frame based on its ID.
+ */
+static void decode_and_print_frame(zynqmp_canFrame *frame)
```

Comment:
Use camelCase, perhaps file prefix? Not that important I guess for a test.
```suggestion
static void zynqmpCanTest_decodeAndPrintFrame(zynqmp_canFrame *frame)
```